logging/bin/remove_eventrouter.sh

```diff
@@ -10,8 +10,23 @@ this_script=`basename "$0"`
 
 log_debug "Script [$this_script] has started [$(date)]"
 
+# Copy template files to temp
+logDir=$TMP_DIR/$LOG_NS
+mkdir -p $logDir
+cp -R logging/eventrouter/eventrouter.yaml $logDir/eventrouter.yaml
+
+# Replace placeholders
+log_debug "Replacing logging namespace for files in [$logDir]"
+  if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
+    sed -i '' "s/__LOG_NS__/$LOG_NS/g" $logDir/eventrouter.yaml
+  else
+    sed -i "s/__LOG_NS__/$LOG_NS/g" $logDir/eventrouter.yaml
+  fi
+
 log_info "Removing Event Router [$(date)]"
-kubectl delete --ignore-not-found -f logging/eventrouter.yaml
+# Remove existing instance of Event Router in the kube-system namespace (if present).
```

Comment:
Is this being done to handle "older" deployments?  Or, do we think we will have clusters where there will be two instances of Event Router running?  I wonder if should only delete one instance (with the one in the $LOG_NS being our first/preferred target)?  If our "new" one is there, what makes us think the one in kube-system is "ours" too?  I'm not pushing for anything, just raising the question.

Reply:
This is currently how I am handling the "older" deployments until I can come up with a more concrete way to identify "our" Event Router.

I haven't implemented the logic needed to stop once after an instance has been found and removed (checking from in $LOG_NS > in kube-systems) ... though, even that could unnecessarily remove an instance in kube-system.

I'm open to any suggestions around this.